src/control/client/service.go

```diff
@@ -24,47 +24,22 @@
 package client
 
 import (
-	"time"
-
 	pb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
-
-	"github.com/pkg/errors"
 	"golang.org/x/net/context"
 )
 
-func (c *control) killRank(uuid string, rank uint32) error {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
-
-	resp, err := c.client.KillRank(ctx, &pb.DaosRank{PoolUuid: uuid, Rank: rank})
-	if err != nil {
-		return err
-	}
-	if resp.Status != pb.DaosRequestStatus_SUCCESS {
-		return errors.New("DAOS request status " + resp.Status.String())
-	}
-
-	return nil
-}
-
 // KillRank Will terminate server running at given rank on pool specified by
-// uuid.
+// uuid. Request will only be issued to a single access point.
 func (c *connList) KillRank(uuid string, rank uint32) ResultMap {
 	results := make(ResultMap)
-	ch := make(chan ClientResult)
+	mc := c.controllers[0]
```

Comment:
I'm curious about the motivation for always selecting the first controller. Will there ever be more than one entry point?

Reply:
good question, kill rank and create pool are what we have sort of defined as "mgmt_ctl_svc" and are forwarded by the MgmtControlService grpc server, when one of these commands is called in the management tool, Rather than broadcasting to all hosts in host-list or host-file it will instead connect to the first access point and this will be available as controllers[0] . in future there will be multiple entry points, @liw can correct me if this statement is inaccurate.